openedx/core/djangoapps/content/learning_sequences/api/processors/schedule.py

```diff
@@ -44,9 +44,10 @@ def __init__(self, course_key: CourseKey, user: types.User, at_time: datetime):
 
     def load_data(self):
         """Pull dates information from edx-when."""
-        # (usage_key, 'due'): datetime.datetime(2019, 12, 11, 15, 0, tzinfo=<UTC>)
-        # TODO: Merge https://github.com/edx/edx-when/pull/48 and add `outline_only=True`
-        self.dates = get_dates_for_course(self.course_key, self.user)
+        # Return data format: (usage_key, 'due'): datetime.datetime(2019, 12, 11, 15, 0, tzinfo=<UTC>)
+        self.dates = get_dates_for_course(
+            self.course_key, self.user, subsection_and_higher_only=True, published_version=self.course_version
```

Comment:
This line activates the optimization at the heart of this change - the only course block dates needed are the ones at the subsection level and higher in the course.